oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java

```diff
@@ -1115,6 +1123,15 @@ private <T extends Document> T findAndModify(Collection<T> collection,
                 }
             }
             return oldDoc;
+        } catch (MongoWriteException e) {
+            WriteError werr = e.getError();
+            LOG.error("Failed to update the document with Id={} with error message '{}'",
```

Comment:
```suggestion
            LOG.error("Failed to update the document with Id={} with a MongoWriteException message '{}'",
```

a bit a detail, but perhaps could be useful to explicitly state that it was a MongoWriteException (and a MongoCommandException below)